### PR TITLE
Switching from RestClient to Faraday

### PR DESCRIPTION
**Pull request by hungryzi**
My main motivation is to add support to NTLM for my personal use and #12.
The secondary motivation is that Faraday looks better maintained than RestClient.

Faraday supports multiple adapters:
- Net::HTTP
- Excon
- Typhoeus
- Patron
- EventMachine
- HTTPClient

I plan to use Typhoeus(https://github.com/typhoeus/typhoeus).

### CODE REVIEW

File: features/step_definitions/service_steps.rb [hunk 1]
```diff
@@ -45,6 +45,9 @@
 end
 
 Then /^I should be able to call "([^\"]*)" on the service$/ do |method|
+  require 'pry'
+  binding.pry
```
Comment (zzk) [inline features/step_definitions/service_steps.rb]: This needs to be removed!

**Comment by zzk** [comment]
@visoft could you point me how to set up sample service with let's say virtualbox?

**Comment by visoft** [comment]
Well first you need to install Windows. I have a Window 7 machine running under Parallels. From there, just get .NET installed. Once you have that, clone `ruby_odata` and run the appropriate batch file for IIS Express. It should fire right up. You'll need to note the endpoint IP and such and make the changes to your configuration as appropriate so you can communicate with the virtual machine. I hope this helps!

**Comment by visoft** [comment]
@zzk I wrote up a detailed [blog post](http://bit.ly/1HucTdh) on setting up VirtualBox for the SampleService, I hope this helps!

**Comment by zzk** [comment]
Awesome!
I figure out that I need iis and ruby install on my win host.

On the other hand - did you ever think about moving from vcr? It's not very ruby way to install win7 with iis to re-record test if there's a need for that. I would advice to switch to fake service made with sinatra - take a look [here](https://robots.thoughtbot.com/how-to-stub-external-services-in-tests)

**Comment by visoft** [comment]
@zzk, all the RSpec tests use that approach, using snapshots of the EDMX metadata and the result. They are completely isolated. The Cucumber features are more integration tests to ensure it works correctly with a live service. Of course these don't work using a CLI like Travis, hence the need for vcr.

**Comment by visoft** [comment]
Closed because of #58

**Pull request closed without merging**